File: update202004/b/answer.py
# 読み間違えで動かない
ballCountN = int(input())
blueBalls = []
redBalls = []
for _ in range(ballCountN):
    number_str, color = input().split()
    if color == 'B':
        blueBalls.append(int(number_str))
    elif color == 'R':
        redBalls.append(int(number_str))
blueBalls.sort()
redBalls.sort()
while True:
    if len(redBalls) > 0:
        print(redBalls.pop(0))
    if len(blueBalls) > 0:
        print(blueBalls.pop(0))
    if len(redBalls) == 0 and len(blueBalls) == 0:
        break

File: update202004/b/answer.kt
// TODO: listとmutableListで解いてみる
fun main(args: Array<String>) {
    val ballCountN = readLine()!!.toInt()
    var blueBalls = mutableListOf<Int>()  // mutableListにする必要なし
    var redBalls = mutableListOf<Int>()
    for (x in 1..ballCountN) {
        val line = readLine()!!.split(' ')
        when (line[1]) {
            "B" -> blueBalls.add(line[0].toInt())
            "R" -> redBalls.add(line[0].toInt())
        }
    }
    blueBalls.sort()
    redBalls.sort()
    for (ball in redBalls) println(ball)
    for (ball in blueBalls) println(ball)
}
